core/domain/question_services.py

```diff
@@ -122,16 +122,26 @@ def get_questions_by_skill_ids(question_count, skill_ids, start_cursor):
             questions are to be returned. This value should be urlsafe.
 
     Returns:
-        list(Question), str. The list of questions which are linked to the given
-            skill ids and the next cursor value to be used for the next
-            batch of questions (or None if no more pages are left). The returned
-            next cursor value is urlsafe.
+        list(dict), str. The list of questions and the corresponding linked
+            skill descriptions which are linked to the given skill ids and the
+            next cursor value to be used for the next batch of questions (or
+            None if no more pages are left). The returned next cursor value is
+            urlsafe.
     """
-    question_ids, next_cursor = (
+    question_skill_link_dicts, next_cursor = (
```

Comment:
The output should be a domain object, not a dict.

Also, the function seems a bit misnamed now? It used to return question IDs as stated on the tin, but now it looks like it is returning something else. We should rename as needed to ensure that the concepts are still being represented correctly.

Reply:
Done, returning the model itself here and creating a domain object in this function.